shared/engine/index.js

```diff
@@ -410,6 +414,7 @@ class Engine {
 
   reset () {
     resetClient()
+    this._setupClient()
```

Comment:
Does this actually work on mobile. Also I'm changing this file a bunch so let's not do too much in here (this is fine but if we need a ton of things maybe we should wait)